hiro.proto

```diff
@@ -2257,3 +2266,87 @@ message SyncResponse {
   // Current server time.
   int64 current_time_sec = 11;
 }
+
+// Represents an available reward based on the progress of a streak.
+message StreakAvailableReward {
+  // Minimum streak progress count required for this reward, inclusive.
+  int64 count_min = 1;
+  // Maximum streak progress count required for this reward, inclusive.
+  int64 count_max = 2;
+  // Available reward contents.
+  AvailableRewards reward = 3;
+}
+
+// A single claimed streak reward.
+message StreakReward {
+  // Minimum streak progress count required for this reward, inclusive.
+  int64 count_min = 1;
+  // Maximum streak progress count required for this reward, inclusive.
+  int64 count_max = 2;
+  // Reward that was actually granted.
+  Reward reward = 3;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when this reward was claimed.
+  int64 claim_time_sec = 4;
+}
+
+// An individual streak, along with its status and progress if any.
+message Streak {
+  // Unique identifier.
+  string id = 1;
+  // Display name for this streak.
+  string name = 2;
+  // Current overall progress count.
+  int64 count = 3;
+  // Maximum allowed overall progress count.
+  int64 max_count = 4;
+  // Progress count submitted during this reset.
+  int64 count_current_reset = 5;
+  // Maximum progress count allowed during any single reset.
+  int64 max_count_current_reset = 6;
+  // How much overall count is lost per idle reset.
+  int64 idle_count_decay_reset = 7;
+  // How much overall count can be lost total in any contiguous idle period.
+  int64 max_idle_count_decay = 8;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when the streak's last reset period was.
+  int64 prev_reset_time_sec = 9;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when the streak's next reset period is.
+  int64 reset_time_sec = 10;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when the streak was first registered for this user.
+  int64 create_time_sec = 11;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when the streak was last updated.
+  int64 update_time_sec = 12;
+  // The UNIX time (for gRPC clients) or ISO string (for REST clients) when the streak was last claimed.
+  int64 claim_time_sec = 13;
+  // All rewards configured for this streak.
```

Comment:
I'm also guessing these are all readonly properties too? As only the progress will be updated from the client via the update request call?

Reply:
Correct, the update or reset calls are how the client submits data to streaks.